src/pudl/settings.py

```diff
@@ -243,41 +244,82 @@ class Eia860Settings(GenericDatasetSettings):
         data_source: DataSource metadata object
         years: list of years to validate.
         eia860m: whether or not to incorporate an EIA-860m month.
-        eia860m_year_month ClassVar[str]: The 860m year-month to incorporate.
+        eia860m_year_months ClassVar[str]: The 860m year-months to incorporate.
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("eia860")
     years: list[int] = data_source.working_partitions["years"]
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    all_eia860m_year_months: list[str] = DataSource.from_id(
+        "eia860m"
+    ).working_partitions["year_months"]
+    eia860m_year_months: list[str] = [max(all_eia860m_year_months)]
 
-    @field_validator("eia860m")
+    @field_validator("eia860m_year_months")
     @classmethod
-    def check_eia860m_year_month(cls, eia860m: bool) -> bool:
-        """Check 860m date-year is exactly one year after most recent working 860 year.
+    def add_other_860m_years(cls, v, info: FieldValidationInfo) -> list[str]:
```

Comment:
I think this might do the right thing:

```py
    @field_validator("eia860m_year_months")
    @classmethod
    def add_other_860m_years(cls, v, info: FieldValidationInfo) -> list[str]:
        """Find extra years from EIA860m if applicable.

        There's a gap in reporting (after the new year but before the EIA 860 early
        release data in June) when we rely on two years worth of EIA860m data instead
        of just one. This function adds the last available month_year values for each
        year of 860m data that is not yet available in 860.

        """
        extra_eia860m_year_months = []
        if info.data["eia860m"]:
            all_eia860m_years = set(
                pd.to_datetime(info.data["all_eia860m_year_months"]).year.unique()
            )
            # The years in 860m that are not in 860
            extra_eia860m_years = {
                year for year in all_eia860m_years if year not in info.data["years"]
            }
            # The years already listed as variables in eia860m_year_months
            years_in_v = set(pd.to_datetime(v).year.unique())
            # The max year_month values available in 860m for each year not
            # covered by EIA860 (and not already listed in the eia860m_year_months
            # variable)
            extra_eia860m_year_months = [
                max(
                    date
                    for date in pd.to_datetime(info.data["all_eia860m_year_months"])
                    if date.year == year
                )
                for year in (extra_eia860m_years - years_in_v)
                if year > max(info.data["years"])
            ]
        return v + list(pd.Series(extra_eia860m_year_months).dt.strftime("%Y-%m"))
```

Reply:
But I'm not sure what the expected / desired behavior is for adding "extra" EIA-860M years. I think the **only** years we ever want to populated withe EIA-860M data are the ones that are not available from EIA-860 at all, rather than just any that weren't included in the arguments to the settings.